[Clang][X86] allow VPERMILPD/S imm intrinsics to be used in constexpr

stomfaig opened the pull request:
adding cases for `vpermilpd` and `vpermilps` in `InterpBultin`.

Resolves https://github.com/llvm/llvm-project/issues/166529

stomfaig committed:
adding cases for vpermilpd and vpermilps

tbaederr commented:
This is missing changes to `ExprConstant.cpp` and tests.

RKSimon reviewed (changes requested):
Still missing ExprConstant handling and test coverage

stomfaig committed:
vpermilps has same impl as pshufd

stomfaig committed:
impl vpermilps and vpermilpd in ExprConstant

stomfaig committed:
format

stomfaig commented:
Sorry for the oversight, I moved vpermilps under pshufd handling and added ExprConstant impl.

In terms of tests, I could only find tests for already implemented intrinsics like _mm_shuffle_epi32 in CodeGen. Am I only to add tests similar to those?

stomfaig committed:
Merge branch 'main' into issue_166529

RKSimon reviewed (changes requested):
test cases - again, _mm_permute_ps etc. can be adapted from existing _mm_shuffle_epi32 sse2/avx2/mask/maskz TEST_CONSTEXPR tests, but _mm_permute_pd etc will need fresh tests

stomfaig committed:
make intrinsics Constexpr

stomfaig committed:
adding tests

RKSimon reviewed (approved):
LGTM - cheers

RKSimon committed:
Merge branch 'main' into issue_166529
